lib/puma/server.rb

```diff
@@ -424,7 +425,7 @@ def process_client(client, buffer)
 
             ThreadPool.clean_thread_locals if clean_thread_locals
 
-            unless client.reset(@status == :run)
+            unless client.reset(@status == :run && @fast_track_ka_timeout)
```

Comment:
Avoid calculating the same thing multiple times, because `@max_threads` is static.